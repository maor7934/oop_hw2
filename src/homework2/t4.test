#check circle:

CreateNode a 10
CreateNode b 10
CreateNode c 10
CreateNode d 10
CreateNode e 10
CreateNode f 10

CreateGraph SmallCircle
CreateGraph BigCircle
CreateGraph MultiCircle

AddNode SmallCircle a
AddNode SmallCircle b
AddNode SmallCircle c

AddEdge SmallCircle a b
AddEdge SmallCircle b c
AddNode SmallCircle c a

ListNodes SmallCircle
FindPath SmallCircle a -> a
FindPath SmallCircle a -> c

#
AddNode BigCircle a
AddNode BigCircle b
AddNode BigCircle c
AddNode BigCircle d
AddNode BigCircle e
AddNode BigCircle f

AddEdge BigCircle a b
AddEdge BigCircle b c
AddNode BigCircle c d
AddNode BigCircle d e
AddNode BigCircle e f
AddNode BigCircle f a

ListNodes BigCircle
FindPath BigCircle a -> a
FindPath BigCircle a -> f

#

AddNode MultiCircle a
AddNode MultiCircle b
AddNode MultiCircle c
AddNode MultiCircle d
AddNode MultiCircle e
AddNode MultiCircle f

AddEdge MultiCircle a b
AddEdge MultiCircle b c
AddNode MultiCircle c a

AddEdge MultiCircle d e
AddEdge MultiCircle e f
AddNode MultiCircle f d

AddNode MultiCircle c d

ListNodes MultiCircle
FindPath MultiCircle a -> a
FindPath MultiCircle a -> f
FindPath MultiCircle f -> a
FindPath MultiCircle f -> f
