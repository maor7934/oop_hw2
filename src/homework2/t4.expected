#check circle:

created node a with cost 10
created node b with cost 10
created node c with cost 10
created node d with cost 10
created node e with cost 10
created node f with cost 10

created graph SmallCircle
created graph BigCircle
created graph MultiCircle

added node a to SmallCircle
added node b to SmallCircle
added node c to SmallCircle

added edge from a to b in SmallCircle
added edge from b to c in SmallCircle
added edge from c to a in SmallCircle

SmallCircle contains: a b c
the children of a in SmallCircle are: b
the children of b in SmallCircle are: c
the children of c in SmallCircle are: a
shortest path in SmallCircle: a
shortest path in SmallCircle: a b c

#
added node a to BigCircle
added node b to BigCircle
added node c to BigCircle
added node d to BigCircle
added node e to BigCircle
added node f to BigCircle

added edge from a to b in BigCircle
added edge from b to c in BigCircle
added edge from c to d in BigCircle
added edge from d to e in BigCircle
added edge from e to f in BigCircle
added edge from f to a in BigCircle

BigCircle contains: a b c d e f
the children of a in BigCircle are: b
the children of f in BigCircle are: a
shortest path in BigCircle: a
shortest path in BigCircle: a b c d e f

#

added node a to MultiCircle
added node b to MultiCircle
added node c to MultiCircle
added node d to MultiCircle
added node e to MultiCircle
added node f to MultiCircle

added edge from a to b in MultiCircle
added edge from b to c in MultiCircle
added edge from c to a in MultiCircle

added edge from d to e in MultiCircle
added edge from e to f in MultiCircle
added edge from f to d in MultiCircle

added edge from c to d in MultiCircle

MultiCircle contains: a b c d e f
the children of a in MultiCircle are: b
the children of f in MultiCircle are: d
shortest path in MultiCircle: a
shortest path in MultiCircle: a b c d e f
no path found in MultiCircle
shortest path in MultiCircle: f
